dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/RpcServiceContext.java

```diff
@@ -201,6 +202,10 @@ public URL getUrl() {
     @Override
     public void setUrl(URL url) {
         this.url = url;
+        if (url.getSide().equals(CommonConstants.CONSUMER)) {
+            this.remoteApplicationName = url.getRemoteApplication();
+            this.setRemoteAddress(url.getHost(), url.getPort());
+        }
```

Comment:
![image](https://github.com/apache/dubbo/assets/9292748/9454d8c5-355b-4d08-851a-c468d114efae)

Set remote attribute in `org.apache.dubbo.rpc.cluster.support.AbstractClusterInvoker#invokeWithContext` would be a better choice.
```
    private static void setRemote(Invoker<?> invoker, Invocation invocation) {
        invocation.addInvokedInvoker(invoker);
        RpcContext.getServiceContext().setRemoteAddress(invoker.getUrl().toInetSocketAddress());
        RpcContext.getServiceContext().setRemoteApplicationName(invoker.getUrl().getRemoteApplication());
    }
```

Reply:
@AlbumenJ hello, I have fix them,and I run the local test on filter  can get them